terraform_providers_lock.sh

```diff
@@ -2,33 +2,53 @@
 
 set -eo pipefail
 
-main() {
-  initialize_
-  parse_cmdline_ "$@"
-  terraform_providers_lock_
+function main {
+  common::initialize
+  common::parse_cmdline "$@"
+  common::per_dir_hook "${ARGS[*]}" "${FILES[@]}"
 }
 
-initialize_() {
+function common::colorify {
+  # Colors. Provided as first string to first arg of function.
+  # shellcheck disable=SC2034
+  local -r red="$(tput setaf 1)"
+  # shellcheck disable=SC2034
+  local -r green="$(tput setaf 2)"
+  # shellcheck disable=SC2034
+  local -r yellow="$(tput setaf 3)"
+  # Color reset
+  local -r RESET="$(tput sgr0)"
+
+  # Params start #
+  local COLOR="${!1}"
+  local -r TEXT=$2
+  # Params end #
+
+  if [ "$PRE_COMMIT_COLOR" = "never" ]; then
```

Comment:
Double square quotes are used most of the time across this file, so to unify approaches used:
```suggestion
  if [[ "$PRE_COMMIT_COLOR" == "never" ]]; then
```